### PR TITLE
[159] Create Contract models

### PR DESCRIPTION
**Pull request by d-a-v-e**
## Changes in this PR

* Adds `NewContract`
* Adds `ExistingContract`

### CODE REVIEW

**Comment by d-a-v-e** [comment]
[Contract data model](https://trello.com/c/rB3N1LzC/159-contract-data-model)

**Review by kennyevil — APPROVED** [review]

File: app/models/support/contract.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Support
```
Comment (AndrewCowle) [inline app/models/support/contract.rb:1]: Can you add some documentation (like the other supported models please) just to explain what it is etc

File: app/models/support/existing_contract.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
```
Comment (AndrewCowle) [inline app/models/support/existing_contract.rb:2]: Just some docs

File: spec/factories/contracts.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :contract do
+    initialize_with { type.present? ? type.constantize.new : Contract.new }
+
+    supplier { "MyString" }
```
Comment (AndrewCowle) [inline spec/factories/contracts.rb:5]: "Example Supplier"

**Review by AndrewCowle — CHANGES_REQUESTED** [review]

**Review by EdwinKruglov — APPROVED** [review]

**Pull request merged**